src/Main.elm

```diff
@@ -25,38 +69,325 @@ init =
 
 type Msg
     = NoOp
+    | EditTag Int
+    | DeleteTag Int
+    | CancelPendingDelete Int
+    | OpenAddForm
+    | CloseAddForm
+    | AddNewTag
+    | InsertNewTagName String
+    | InsertNewTagValue String
+    | InsertEditTagValue String
+    | SubmitEditTagValue Int
 
 
 update : Msg -> Model -> ( Model, Cmd Msg )
-update _ model =
-    ( model, Cmd.none )
+update msg model =
+    case msg of
+        NoOp ->
+            ( model, Cmd.none )
 
+        EditTag index ->
+            case List.Extra.getAt index model.tags of
+                Just tag ->
+                    ( { model | editingTagIndex = Just index, editingTagValue = Maybe.withDefault "" tag.value, ariaLiveMessage = Just ("Editing tag " ++ tag.name) }
+                    , Task.attempt (\_ -> NoOp) (Dom.focus "edit-tag-value-input")
+                    )
 
+                Nothing ->
+                    ( model, Cmd.none )
 
----- VIEW ----
+        DeleteTag index ->
+            case model.pendingDeleteOnIndex of
+                Just pendingIndex ->
+                    if pendingIndex == index then
+                        if Just index == model.editingTagIndex then
+                            ( { model
+                                | tags = List.Extra.removeAt index model.tags
+                                , editingTagIndex = Nothing
+                                , editingTagValue = ""
+                                , ariaLiveMessage = Just "Tag deleted successfully"
+                                , pendingDeleteOnIndex = Nothing
+                              }
+                            , Cmd.none
+                            )
+
+                        else
+                            ( { model
+                                | tags = List.Extra.removeAt index model.tags
+                                , ariaLiveMessage = Just "Tag deleted successfully"
+                                , pendingDeleteOnIndex = Nothing
+                              }
+                            , Cmd.none
+                            )
+
+                    else
+                        ( { model
+                            | pendingDeleteOnIndex = Just index
+                            , ariaLiveMessage = Just "Click delete again to confirm"
+                          }
+                        , Task.perform (\_ -> CancelPendingDelete index)
+                            (Task.succeed ()
+                                |> Task.andThen (always (Process.sleep 3000))
+                            )
+                        )
```

Comment:
These nested ifs are liiitle bit hard to read. Any idea how you'd make this part of the code better?